pkg/http/core/kubernetes/deploy.go

```diff
@@ -155,12 +155,20 @@ func Deploy(c *gin.Context, dm DeployManifestRequest) {
 				},
 			}
 
+			ls, err := metav1.LabelSelectorAsSelector(&labelSelector)
+			if err != nil {
+				clouddriver.Error(c, http.StatusInternalServerError, err)
+				return
+			}
+
 			lo := metav1.ListOptions{
-				LabelSelector:  labels.Set(labelSelector.MatchLabels).String(),
+				LabelSelector:  ls.String(),
```

Comment:
What is the output of this label selector string? Have we verified that it is only pulling down resources that have a version annotation?

Reply:
No, I can add a test for that